tensorly/decomposition/_parafac2.py

```diff
@@ -303,9 +367,10 @@ def parafac2(
     if absolute_tol is None:
         absolute_tol = tl.eps(factors[0].dtype) * 1000
 
-    acc_pow = 2.0  # Extrapolate to the iteration^(1/acc_pow) ahead
-    acc_fail = 0  # How many times acceleration have failed
-    max_fail = 4  # Increase acc_pow with one after max_fail failure
+    if linesearch:
+        line_search = _bro_thesis_line_search(
+            norm_tensor, svd, verbose=verbose, nn_modes=nn_modes
+        )
```

Comment:
```suggestion
if linesearch and not isinstance(linesearch, _BroThesisLineSearch):
    linesearch = _BroThesisLineSearch(
        norm_tensor, svd, verbose=verbose, nn_modes=nn_modes
    )
```

It could be nice to support sending in a line search object (keeping this option undocumented so it's not officially a part of the public API yet). Then we also don't need a separate variable for the `_BroThesisLineSearch` instance. What do you think?